### PR TITLE
QPID-8472:[Broker-J]Improve operational logging for operations on queue

### PR DESCRIPTION
**Pull request by Dedeepya-T**
Improve operational logging for operations on queue

### CODE REVIEW

**Review by alex-rufous — CHANGES_REQUESTED** [review]
The operation logging can also be added for clearQueue operation

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 1]
```diff
@@ -3520,23 +3520,52 @@ public MessageEnqueueRecord getEnqueueRecord()
                                                                           destination,
                                                                           parseSelector(selector),
                                                                           limit);
-        _virtualHost.executeTransaction(transaction);
+        performOperationWithLogging("Copy",destination, transaction);
         return transaction.getModifiedMessageIds();
 
     }
 
+    private void performOperationWithLogging(final String msg,final Queue<?> destination, final QueueSizeLimitRespectingTransaction transaction)
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3528]: Done!

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 2]
```diff
@@ -3520,23 +3520,52 @@ public MessageEnqueueRecord getEnqueueRecord()
...
     @Override
     public List<Long> deleteMessages(final List<Long> messageIds, final String selector, int limit)
     {
         DeleteMessagesTransaction transaction = new DeleteMessagesTransaction(this,
                                                                               messageIds,
                                                                               parseSelector(selector),
                                                                               limit);
-        _virtualHost.executeTransaction(transaction);
-
+        String status = "FAILED";
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3553]: Refactored to use the common method

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 3]
```diff
@@ -3520,23 +3520,52 @@ public MessageEnqueueRecord getEnqueueRecord()
...
                                                                           limit);
-        _virtualHost.executeTransaction(transaction);
+        performOperationWithLogging("Copy",destination, transaction);
         return transaction.getModifiedMessageIds();
 
     }
 
+    private void performOperationWithLogging(final String msg,final Queue<?> destination, final QueueSizeLimitRespectingTransaction transaction)
+    {
+        String status = "FAILED";
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3530]: Agreed and Done!

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 4]
```diff
@@ -3520,23 +3520,52 @@ public MessageEnqueueRecord getEnqueueRecord()
...
+        try
+        {
+            _virtualHost.executeTransaction(transaction);
+            status = "SUCCESS";
+        }
+        finally
+        {
+            logOperation(String.format(msg+"Messages :source-%s:target-%s:messagecount-%d:status-%s",
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3538]: Done!

**Review by alex-rufous — CHANGES_REQUESTED** [review]

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 5]
```diff
@@ -3520,23 +3521,46 @@ public MessageEnqueueRecord getEnqueueRecord()
                                                                           destination,
                                                                           parseSelector(selector),
                                                                           limit);
-        _virtualHost.executeTransaction(transaction);
+        executeTransactionAndLogResults("copyMessages", destination.getName(), transaction);
         return transaction.getModifiedMessageIds();
 
     }
 
+    private void executeTransactionAndLogResults(final String msg,
```
Comment (alex-rufous) [inline broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3529]: The entire method implementation can be simplified to something like

`

try
{
    _virtualHost.executeTransaction(transaction);
}
finally
{
    logOperation(String.format("%s  : %s: %s: %d",
                                      operationName,
                                       getName(),
                                       ofNullable(destinationName).orElse(""),
                                       transaction.getModifiedMessageIds().size()));
}
`

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 6]
```diff
@@ -3520,23 +3521,46 @@ public MessageEnqueueRecord getEnqueueRecord()
...
     @Override
     public List<Long> deleteMessages(final List<Long> messageIds, final String selector, int limit)
     {
         DeleteMessagesTransaction transaction = new DeleteMessagesTransaction(this,
                                                                               messageIds,
                                                                               parseSelector(selector),
                                                                               limit);
-        _virtualHost.executeTransaction(transaction);
-
+        executeTransactionAndLogResults("deleteMessages",null, transaction);
```
Comment (alex-rufous) [inline broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3561]: space is missed between parameters

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java (hunk 5 above)
Comment (alex-rufous) [inline broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:3529]: The parameter name 'msg' needs to be changed to operationName
Let move transaction parameter to a second position
the destinationName can be renamed into a targetdestinationName

**Review by alex-rufous — CHANGES_REQUESTED** [review]

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 7]
```diff
@@ -1826,7 +1827,7 @@ public long clearQueue()
         }
 
         txn.commit();
-
+        executeTransactionAndLogResults("clearQueue",null,null);
```
[the remark this answers is not in the document]
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java]: Applied the simpler fix

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java (hunk 6 above)
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java]: This method is removed!

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java (hunk 5 above)
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java]: As per another comment this method is removed

Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java]: this method no longer exists!

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java [hunk 8]
```diff
@@ -1826,7 +1826,7 @@ public long clearQueue()
         }
 
         txn.commit();
-
+        logOperation(String.format("clearQueue : %s", getName()));
```
Comment (alex-rufous) [inline broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java:1829]: Let's add count

**Review by alex-rufous — CHANGES_REQUESTED** [review]

File: broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java (hunk 8 above)
Comment (Dedeepya-T) [reply broker-core/src/main/java/org/apache/qpid/server/queue/AbstractQueue.java]: Done!

**Pull request closed without merging**